app/scripts/directives/uiAceYaml.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('uiAceYaml', {
+    controller: [
+      '$scope',
+      UIAceYAML
+    ],
+    controllerAs: '$ctrl',
+    bindings: {
+      resource: '=',
+      ngRequired: '<?',
+      showFileInput: '<?'
+    },
+    templateUrl: 'views/directives/ui-ace-yaml.html'
+  });
+
+  function UIAceYAML($scope) {
+    var ctrl = this;
+    var aceEditor;
+
+    var parseYAML = function(strict) {
+      // https://github.com/nodeca/js-yaml#safeload-string---options-
+      return jsyaml.safeLoad(ctrl.model, {
+        // If `strict` is false, allow duplicate keys in the YAML for
+        // compability with `oc create` using the js-yaml `json` option. This
+        // also lets us parse JSON as YAML, where duplicate keys are allowed.
+        json: !strict
+      });
+    };
+
+    var clearAnnotations = function() {
+      aceEditor.getSession().clearAnnotations();
+      $scope.$evalAsync(function() {
+        ctrl.annotations = {};
+      });
+    };
+
+    var setAnnotation = function(e, severity) {
+      var session = aceEditor.getSession();
+      var length = session.getLength();
+      var row = _.get(e, 'mark.line', 0);
+      var col = _.get(e, 'mark.column', 0);
+      var message = e.message || 'Could not parse YAML.';
```

Comment:
Good point. "Could not parse content." maybe?

Hopefully you never see the default message